org.eclipse.xtext/src/org/eclipse/xtext/workspace/ISourceFolder.java

```diff
@@ -20,4 +32,28 @@
 	 * separator. It will never be null.
 	 */
 	URI getPath();
+
+	/**
+	 * @param uri
+	 *            to check
+	 * @return true iff the given {@link URI} is a sub element of the {@link URI} of this {@link ISourceFolder}.
+	 */
+	default boolean contains(URI uri) {
+		URI path = getPath();
+		return UriUtil.isPrefixOf(path, uri);
+	}
+
+	/** @return a list of all URIs that are passed to the acceptor of {@link #scan(URI, IAcceptor)} */
+	default List<URI> getAllResources() {
+		List<URI> uris = new LinkedList<>();
+		scan(uri -> uris.add(uri));
+
+		return uris;
+	}
+
+	/** Scans this source folder and passes all results to the acceptor. */
+	default void scan(IAcceptor<URI> acceptor) {
+		FILE_SYSTEM_SCANNER.scan(getPath(), acceptor);
```

Comment:
Probably something like `SkippableAcceptor` with `skipped=true` which is the default. Clients can then still use another default?